### PR TITLE
Move reagent documentation to dmdoc.

### PR DESCRIPTION
**Pull request by warriorstar-orion**
## What Does This PR Do
This PR migrates the file `code/modules/reagent/chemistry/readme.dm` from a file consisting of a single large comment to the appropriate places in the reagent codebase in dmdoc format. It's not a 1:1 copypaste, I tried to remove useless wordage and got rid of the frankly pretty useless ASCII diagram.

## Why It's Good For The Game
Even if dmdoc doesn't see much usage, having this documentation closer to the relevant code (and showing up in vsc tooltips) is beneficial. Getting rid of a file that's effectively empty is good too.

### CODE REVIEW

**Review by Sirryan2002 — APPROVED** [review]
this is probably better than how its currently implemented

File: code/modules/reagents/chemistry/holder.dm [hunk 1]
```diff
@@ -124,7 +149,17 @@
...
 
-/datum/reagents/proc/trans_to(target, amount = 1, multiplier = 1, preserve_data = TRUE, no_react = FALSE) //if preserve_data=0, the reagents data will be lost. Usefull if you use data for some strange stuff and don't want it to be transferred.
+/**
+ * Equally transfer the contents of the holder to another objects holder.
+ *
+ * You need to pass it the object (not the holder) you want to transfer to and
+ * the amount you want to transfer. Its return value is the actual amount
+ * transfered (if one of the objects is full/empty).
+ *
+ * If `preserve_data=0`, the reagents data will be lost. Useful if you use data
```
Comment (SteelSlayer) [inline code/modules/reagents/chemistry/holder.dm:159]:
```suggestion
 * If `preserve_data = FALSE`, the reagents data will be lost. Useful if you use data
```

File: code/modules/reagents/chemistry/recipes.dm [hunk 2]
```diff
@@ -3,13 +3,26 @@
...
+	// but if you want to, you can use them for other things
 
+	/// The container required for the reaction to happen.
+	/// Leave this null if you want the reaction to happen anywhere.
+	var/atom/required_container = null
+	/// Extra requirements for the reaction to happen.
+	var/required_other = FALSE
+
+	/// This is the amount of the resulting reagent this recipe will produce.
+	/// It's recommended you set this to the total volume of all required reagent.
```
Comment (SteelSlayer) [inline code/modules/reagents/chemistry/recipes.dm:25]:
```suggestion
	/// It's recommended you set this to the total volume of all required reagents.
```

**Review by SteelSlayer — COMMENTED** [review]
Small stuff.

**Pull request merged**